### PR TITLE
Stats: Add stats subscription to My Plan

### LINKED ISSUE #31713: Paid Stats: show Stats purchase in Jetpack Dashboard -> My Plan
Image

### PR DESCRIPTION
**Pull request by kangzj**
Fixes #31713 

## Proposed changes:

* Show Stats subscription in My Plan

### Other information:

- [x] Have you written new tests for your changes, if applicable?
- [x] Have you checked the E2E test CI results, and verified that your changes do not break them?
- [x] Have you tested your changes on WordPress.com, if applicable (if so, you'll see a generated comment below with a script to run)?

## Jetpack product discussion
n/a

## Does this pull request change what data or activity we track or use?
no

## Testing instructions:

* Purchase Stats for your site `https://wordpress.com/checkout/${siteId}/jetpack_stats_monthly`
* Build Jetpack `jetpack build plugins/jetpack`
* Open `/wp-admin/admin.php?page=jetpack#/my-plan`
* Ensure you see Stats / Stats Free listed in My Products

image

### CODE REVIEW

**Comment by ederrengifo** [comment]
@kangzj For the icon we are using the chart icon from the WordPress library, just in case is faster to just reuse it, otherwise, I attach the SVG

chart

**Comment by kangzj** [comment]
Hi @ederrengifo thanks for providing the Stats icon 👍 

image

It seems to me that it doesn't quite match the other icons. Do you think a dark background one would look better?

**Comment by kangzj** [comment]
Hi @ederrengifo I had a play with the SVG and how does this look? Feel free to discard or provide another version if this doesn't look right. 

https://github.com/Automattic/jetpack/blob/7043b718ea588bac96ebecd65d2ab9f2d0cbe5a8/projects/plugins/jetpack/images/products/product-jetpack-stats.svg

image

**Review by jsnmoon — CHANGES_REQUESTED** [review]
These changes look reasonable, but I noticed that the annual free plan gets stuck in a loading state:
image

The annual PWYW and monthly commercial plans work as expected:
image

Aside: I also noticed some unexpected behavior in purchasing the paid stats plans that I've noted in pbNhbs-6ZT-p2#comment-16609.

**Comment by jsnmoon** [comment]
> Hi [Eder] I had a play with the SVG and how does this look? Feel free to discard or provide another version if this doesn't look right.

The updated icon looks great, by the way!

File: projects/plugins/jetpack/_inc/client/my-plan/my-plan-header/index.js [hunk 1]
```diff
@@ -445,6 +445,22 @@ class MyPlanHeader extends React.Component {
...
 
+			case 'is-jetpack-stats-plan':
+				return {
+					...productProps,
+					details: [ activation, expiration ],
+					tagLine: __( 'Simple, yet powerful analytics with priority support.', 'jetpack' ),
+					title: __( 'Jetpack Stats', 'jetpack' ),
+				};
+
+			case 'is-jetpack-stats-free-plan':
```
Comment (kangzj) [inline projects/plugins/jetpack/_inc/client/my-plan/my-plan-header/index.js]:
```suggestion
			case 'is-free-jetpack-stats-plan':
```
Fix typo

**Comment by kangzj** [comment]
@jsnmoon good catch. There was a last minute change which caused a mismatch. Fixed: https://github.com/Automattic/jetpack/pull/31732/commits/fd4b2fa97c5e63c6eb050be33aeae652241ac11a

**Review by jsnmoon — APPROVED** [review]
Looks solid!

**Pull request merged**